components/script/dom/stylesheet.rs

```diff
@@ -44,6 +46,11 @@ impl StyleSheetMethods for StyleSheet {
         self.href.clone()
     }
 
+    fn GetOwnerNode(&self) -> Option<DomRoot<Element>> {
+        let owner = self.downcast::<CSSStyleSheet>().unwrap().get_owner();
```

Comment:
This is not ok. This needs to return `None` if you have removed the stylesheet from the element somehow.

For example:

```html
<!doctype html>
<style></style>
<script>
  let sheet = document.querySelector("style").sheet;
  console.log(sheet.ownerNode); // Should log "[Object HTMLStyleElement]".
  sheet.ownerNode.remove();
  console.log(sheet.ownerNode); // Should log null.
</script>
```

Reply:
Oh yes, I''ll correct it. Thanks for pointing it out.

Reply:
How should I go about? Should I change the owner in CSSStyleSheet to Option?  Or is there any simpler way to detect null?

Reply:
Yes, I think that's the best thing to do. You need to clear the owner pointer when detaching the stylesheet.

Reply:
So, will I have to change the implementation of the get_owner function too?

Reply:
Btw, I think it might be worth making the above test case into a WPT test? 👀

Reply:
Yes, if there's no WPT test testing this already then we should add it.

Reply:
>  So, will I have to change the implementation of the get_owner function too? From &Element to Option<DomRoot>? Or something else?

Yeah, probably should return an `Option<&Element>`

Reply:
@sbansal3096 In case that you didn't see the messages in IRC, you can the log here 👀
https://mozilla.logbot.info/servo/20190326#c16133761